pkg/security/ptracer/ptracer.go

```diff
@@ -197,11 +240,36 @@ func (t *Tracer) Trace(cb func(cbType CallbackType, nr int, pid int, ppid int, r
 			break
 		}
 
-		if waitStatus.Exited() || waitStatus.Signaled() {
+		if waitStatus.Exited() {
+			if pid == t.PID {
+				break
+			}
+			cb(CallbackExitType, ExitNr, pid, 0, regs, &ebpfless.ExitSyscallMsg{
```

Comment:
I would rather pass the waitStatus to the Callback and put the logic in the cws.go file CoreDump/Signaled/Exited.

Reply:
Suggestion applied